client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -34,35 +35,69 @@ const OnboardingEdit: FunctionComponent< BlockEditProps< Attributes > > = () =>
 			{ isCreatingSite && (
 				<Redirect push={ replaceHistory ? undefined : true } to={ makePath( Step.CreateSite ) } />
 			) }
-			<Switch>
-				<Route exact path={ makePath( Step.IntentGathering ) }>
-					<AcquireIntent />
-				</Route>
+			<Route exact path={ makePath( Step.IntentGathering ) }>
+				{ ( { match } ) => (
+					<CSSTransition
+						in={ match !== null }
+						unmountOnExit
+						classNames="gutenboarding-page"
+						timeout={ 500 }
+					>
+						<AcquireIntent />
+					</CSSTransition>
+				) }
+			</Route>
 
+			{ ! siteVertical && (
 				<Route path={ makePath( Step.DesignSelection ) }>
-					{ ! siteVertical ? (
-						<Redirect to={ makePath( Step.IntentGathering ) } />
-					) : (
-						<DesignSelector />
-					) }
+					<Redirect to={ makePath( Step.IntentGathering ) } />
 				</Route>
+			) }
+
+			<Route path={ makePath( Step.DesignSelection ) }>
+				{ ( { match } ) => (
+					<CSSTransition
+						in={ match !== null }
+						unmountOnExit
+						classNames="gutenboarding-page"
+						timeout={ 500 }
+					>
+						<DesignSelector />
+					</CSSTransition>
+				) }
+			</Route>
 
+			{ ! selectedDesign && ! isEnabled( 'gutenboarding/style-preview' ) && (
 				<Route path={ makePath( Step.Style ) }>
-					{ // Disable reason: Leave me alone, my nested ternaries are amazing ✨
-					// eslint-disable-next-line no-nested-ternary
-					! selectedDesign ? (
-						<Redirect to={ makePath( Step.DesignSelection ) } />
-					) : isEnabled( 'gutenboarding/style-preview' ) ? (
-						<StylePreview />
-					) : (
-						<Redirect to={ makePath( Step.DesignSelection ) } />
-					) }
+					<Redirect to={ makePath( Step.DesignSelection ) } />
 				</Route>
+			) }
 
-				<Route path={ makePath( Step.CreateSite ) }>
-					<CreateSite />
-				</Route>
-			</Switch>
```

Comment:
Not sure if this is related to removing the `<Switch />` or it invisible without the transitions. When redirecting, it seems that we are first loading the original component, then transitioning to the new one which results in a not so nice effect and possible errors (eg: some components might need variables that are not yet defined)

Reply:
I see. This gives more points to #40977